rosbag2_storage/include/rosbag2_storage/filesystem_helper.hpp

```diff
@@ -141,6 +142,24 @@ class FilesystemHelper
     struct stat stat_buffer {};
     return stat(file_path.c_str(), &stat_buffer) == 0;
   }
+
+  /**
+   * Creates the directory p as if by POSIX mkdir() with a second argument of 0777.
+   * The parent directory must already exist.
+   * @param p the path to the new directory to create.
+   * @return true if a directory was created for the directory p.
+   */
+  static bool create_directory(const std::string & p)
+  {
+#ifdef _WIN32
+    _mkdir(p.c_str());
+#else
+    constexpr const int perms_all = S_IRWXU | S_IRWXG | S_IRWXO;  // same as 0777
```

Comment:
so I believe Unix systems create folders with 755 by default. Why 777?

Reply:
`std::filesystem::create_directory` documentation says it uses 
```c++
static_cast<int>(std::filesystem::perms::all)
```
which maps to 777

Reply:
The design choise to check `file_exists` on return is to match up with C++17's `std::filesystem::create_directory` which returns True if the directory is created.